motorutil/steer.go

```diff
@@ -0,0 +1,232 @@
+// Copyright ©2016 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package motorutil
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ev3go/ev3dev"
+)
+
+// Steering implements a paired-motor steering unit similar to an EV-G steering block.
+type Steering struct {
+	// Left and Right are the left and right motors to be
+	// used by the steering module.
+	Left, Right *ev3dev.TachoMotor
+
+	// Timeout is the timeout for waiting for motors to
+	// return to a non-driving state.
+	//
+	// See ev3dev.Wait documentation for timeout behaviour.
+	Timeout time.Duration
+}
+
+// SteerCounts steers in the given turn for the given tacho counts and at the
```

Comment:
It can't be rotations unless I use a float64 since otherwise it precludes part rotation. I can add a comment in the documentation that the counts/rot can be obtained with a call to `t.CountsPerRot` for each motor, but I'd rather not put those calls into the `SteerCounts` method since the value can be cached by the user after an initial call (and as you say counts ≅ degrees).

Reply:
It would be nicer if this library cached the CountsPerRot value and did the conversion so that the user doesn't have to.

Reply:
That was a conscious design decision; none of the devices in ev3go save state that is held by the underlying driver since it may become invalid.

Reply:
Ack. But, `address`, `commands`, `count_per_rot`, `count_per_m`, `full_travel_count`, `driver_name`, `max_speed` and `stop_actions` will never change, so they should be safe to read once and cache.

Reply:
Yes, I think you might be right. The only case where those would be invalidated is if the motor is unplugged and replugged, but then the ID is also invalid, so all bets are off.

This is #53.